api/src/location/location.module.ts

```diff
@@ -1,15 +1,28 @@
-import { Module } from '@nestjs/common';
+import { Module, Provider } from '@nestjs/common';
 import { ScheduleModule } from '@nestjs/schedule';
-
-import { LocationService } from './location.service';
-import { LocationController } from './location.controller';
 import { TypeOrmModule } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
 import { LocationEntity } from './entities/location.entity';
 import { LocationFetchScheduleService } from './location-fetch-schedule.service';
+import { LocationController } from './location.controller';
+import { LocationService } from './location.service';
+
 
 @Module({
   imports: [TypeOrmModule.forFeature([LocationEntity]), ScheduleModule.forRoot()],
   controllers: [LocationController],
-  providers: [LocationService, LocationFetchScheduleService],
+  providers: [
+    LocationService,
+    ((): Provider => {
+      if (!process.env.MAP_BOX_TOKEN) {
+        return {
+          provide: LocationFetchScheduleService,
+          useFactory: () => undefined
+        }
+      } else {
+        return LocationFetchScheduleService
+      }
+    })()
+  ]
```

Comment:
curious what was going on here

Reply:
I check the nest document, 
maybe we could just use 
```
{
 provide: LocationFetchScheduleService,
 useFactory: !process.env.MAP_BOX_TOKEN ? undefined : LocationFetchScheduleService
 }
```

Reply:
@michel-fresh my understanding is to check if `MAP_BOX_TOKEN` exists, if so, then include the service, 
if not, then do nothing.

Correct me if I'm wrong.

Reply:
I tried those approaches above @jerryfreshworks and they each failed for different reasons. Weirdly, the approach I've used seemed to be the simplest. Though, please check it out for yourself if you are curious. Possibly I am not enough of a nest guru to get it working this way.

Reply:
And yeah, to answer @michel-fresh 's question, Jerry is right. It's a conditionally injected service. Seemed like the simplest approach.